diff --git a/maxlength.module b/maxlength.module
index 47acde2..10005b7 100644
--- a/maxlength.module
+++ b/maxlength.module
@@ -20,18 +20,6 @@ function maxlength_help($path, $arg) {
 }
 
 /**
- * Implements hook_permission().
- */
-function maxlength_permission() {
-  return array(
-    'bypass maxlength' => array(
-      'title' => t('Bypass maxlength setting'),
-      'description' => t("If a maxlength is configured for a certain field, the user don't see the value."),
-    ),
-  );
-}
-
-/**
  * Implements hook_element_info_alter().
  */
 function maxlength_element_info_alter(&$cache) {
@@ -167,7 +155,7 @@ function _maxlength_children(&$element, $ms_elements) {
     return;
   }
   elseif (!isset($bypass)) {
-    $bypass = $GLOBALS['user']->uid == 1 ? !variable_get('maxlength_always_for_uid1', FALSE) : user_access('bypass maxlength');
+    $bypass = false;
   }
 
   $children = element_get_visible_children($element);
